cognite/client/_api/simulators/models.py

```diff
@@ -0,0 +1,444 @@
+
+        return self._create_multiple(
+            list_cls=SimulatorModelRevisionList,
+            resource_cls=SimulatorModelRevision,
+            items=revision,
+            input_resource_cls=SimulatorModelRevisionWrite,
+        )
+
+
+class SimulatorModelsAPI(APIClient):
+    _RESOURCE_PATH = "/simulators/models"
+
+    def __init__(self, config: ClientConfig, api_version: str | None, cognite_client: CogniteClient) -> None:
+        super().__init__(config, api_version, cognite_client)
+        self.revisions = SimulatorModelRevisionsAPI(config, api_version, cognite_client)
+        self._warning = FeaturePreviewWarning(
+            api_maturity="General Availability", sdk_maturity="alpha", feature_name="Simulators"
+        )
+        self._RETRIEVE_LIMIT = 1
+        self._CREATE_LIMIT = 1
+        self._DELETE_LIMIT = 1
+
+    def list(
+        self,
+        limit: int = DEFAULT_LIMIT_READ,
+        filter: SimulatorModelsFilter | dict[str, Any] | None = None,
+        sort: CreatedTimeSort | None = None,
+    ) -> SimulatorModelList:
+        """`Filter simulator models <https://developer.cognite.com/api#tag/Simulator-Models/operation/filter_simulator_models_simulators_models_list_post>`_
+        Retrieves a list of simulator models that match the given criteria
+        Args:
+            limit (int): Maximum number of results to return. Defaults to 25. Set to -1, float(“inf”) or None to return all items.
+            filter (SimulatorModelsFilter | dict[str, Any] | None): Filter to apply.
+            sort (CreatedTimeSort | None): The criteria to sort by.
+        Returns:
+            SimulatorModelList: List of simulator models
+
+        Examples:
+            List simulator models:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.models.list()
+
+            Specify filter and sort order:
+                >>> from cognite.client.data_classes.simulators.filters import SimulatorModelsFilter
+                >>> from cognite.client.data_classes.simulators.models import CreatedTimeSort
+                >>> res = client.simulators.models.list(
+                ...     filter=SimulatorModelsFilter(simulator_external_ids=["simulator_external_id"]),
+                ...     sort=CreatedTimeSort(order="asc")
+                ... )
+
+        """
+        self._warning.warn()
+        return self._list(
+            method="POST",
+            limit=limit,
+            resource_cls=SimulatorModel,
+            list_cls=SimulatorModelList,
+            sort=[CreatedTimeSort.load(sort).dump()] if sort else None,
+            filter=filter.dump(camel_case=True) if isinstance(filter, CogniteFilter) else filter,
+        )
+
+    @overload
+    def retrieve(self, id: int | None = None, external_id: str | None = None) -> SimulatorModel | None: ...
```

Comment:
This overload allows "if single id and single external id" -> "SimulatorModel is returned".

I think you actually need to break this into 3, something like this:

```py
from typing import NoReturn

@overload
def retrieve(self, id: None = None, external_id: None = None) -> NoReturn: ...

@overload
def retrieve(self, id: int, external_id: None = None) -> SimulatorModel | None: ...

@overload
def retrieve(self, id: None = None, external_id: str) -> SimulatorModel | None: ...
```